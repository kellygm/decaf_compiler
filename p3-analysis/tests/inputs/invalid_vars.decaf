void i;
bool b;
int arr[0];
int main;
def int main()
{
    int b;
    b = 2;
}