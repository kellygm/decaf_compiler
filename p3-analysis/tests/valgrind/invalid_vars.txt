==2557894== Memcheck, a memory error detector
==2557894== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==2557894== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==2557894== Command: ../decaf inputs/invalid_vars.decaf
==2557894== 
Invalid 'void' variable 'i' on line '1'
Invalid array variable 'arr' of size 0 on line '3'
Invalid variable with name 'main' on line '4'
==2557894== 
==2557894== HEAP SUMMARY:
==2557894==     in use at exit: 0 bytes in 0 blocks
==2557894==   total heap usage: 1,905 allocs, 1,905 frees, 226,224 bytes allocated
==2557894== 
==2557894== All heap blocks were freed -- no leaks are possible
==2557894== 
==2557894== For lists of detected and suppressed errors, rerun with: -s
==2557894== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
