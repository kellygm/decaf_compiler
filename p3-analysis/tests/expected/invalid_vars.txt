Invalid 'void' variable 'i' on line '1'
Invalid array variable 'arr' of size 0 on line '3'
Invalid variable with name 'main' on line '4'
